.github/workflows/ci.yml

```diff
@@ -46,8 +46,8 @@ jobs:
           - "1.18"
           - "1.20"
           - "1.21"
-          - "1.22"
-          - "^1"
+          - "oldstable"
+          - "stable"
```

Comment:
This is somewhat problematic, here's why.

A cache that actions/setup-go creates adds a Go version to the source data for the cache key (or maybe to the cache key itself -- I don't remember exactly). Now, if you change the go version manually (say from 1.22 to 1.23), the existing cache is instantly invalidated. But a string like "stable" or "oldstable" never changes, so when a new minor Go version is released, your CI jobs can still be stuck using an old one from the cache.

The same problem exists when a new Go patch version is released, and it is mitigated by setting the `check-latest: true` input for `action/setup-go`. I believe it also helps when you use `stable/oldstable`,  and in this case it becomes a must-have, if you really want to test Go 1.24.0 once it comes out, not when a cache expires.

I might not be entirely correct here but at the very least it makes sense to keep an eye on your CI runs.

Reply:
Ah, damn. I assumed this was more like a thin alias that wouldn't cause those kinds of issues. I'll add `check-latest: true`, hopefully that mitigates the risk.

Reply:
It seems that the aliases are supposed to imply `check-latest: true` according to the docs:

> Note: using these aliases will result in same version as using corresponding minor release with check-latest input set to true

But I'll add it anyway.